game-app/game-core/src/main/java/games/strategy/triplea/ui/panels/map/MapPanel.java

```diff
@@ -742,7 +743,9 @@ public void paint(final Graphics g) {
     }
     final var unitImageFactory = uiContext.getUnitImageFactory();
     for (final Collection<Unit> value : highlightedUnits) {
-      for (final UnitCategory category : UnitSeparator.categorize(value)) {
+      List<UnitCategory> unitCategories = new ArrayList(UnitSeparator.categorize(value));
+      unitCategories.sort(getComparatorUnitCategories(gameData));
```

Comment:
There is a little bit of a pattern here. Perhaps this can be cleaned up to be something like:
```List<unitCategory> unitCategories = UnitSeparator.sortByUnitType(value)```

Which in turn would do the list copy, categorize, and sorting.
FWIW, there is a bit of a feature envy code smell here where UnitSeparator is used to generate a return value, more data is obtained from UnitSeparator to then operate on that same data again. Resolving 'feature envy code smell' is usually done by moving the functionality entirely into `UnitSeparator`

Reply:
Agree, I create a fix to introduce another method `UnitSeparator.getSortedUnitCategories`.